### PR TITLE
sbcl: update to 2.3.9; fix build on macOS 14

### PR DESCRIPTION
**Pull request by catap**
#### Description

I made an assumption that ECL works well on macOS 14, and uses it to bootstrap SBCL on that system.

Closes: https://trac.macports.org/ticket/68271

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.8 21G725 x86_64
Xcode 14.2 14C18

also tested on the same sysmtem bootstrap via ECL.

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/sbcl-bootstrap/Portfile [hunk 1]
```diff
@@ -20,6 +20,9 @@ homepage            http://www.sbcl.org
 
 use_bzip2           yes
 
+# See: https://trac.macports.org/ticket/68271
+platforms           {darwin < 22}
```
Comment (ryandesign) [inline lang/sbcl-bootstrap/Portfile:24]: Darwin 22 is Ventura. Is this problem there as well, or did you mean to say 23 for Sonoma?

File: lang/sbcl/Portfile [hunk 2]
```diff
@@ -172,8 +130,25 @@ use_parallel_build  no
 build.cmd       sh
 build.target    make.sh
 build.args-append \
-                --prefix=${prefix} \
+                --prefix=${prefix}
+
+# unfortunately sbcl-bootstrap doesn't support macOS Sonoma yet, use ECL instead
+# See: https://trac.macports.org/ticket/68271
+if {${os.major} >= 22} {
```
Comment (ryandesign) [inline lang/sbcl/Portfile:137]: Same question here. Also, check ${os.platform} before checking ${os.major}.

File: lang/sbcl-bootstrap/Portfile (hunk 1 above)
Comment (catap) [reply lang/sbcl-bootstrap/Portfile:24]: Oops, thanks for catching that

File: lang/sbcl/Portfile (hunk 2 above)
Comment (catap) [reply lang/sbcl/Portfile]: Same here

**Comment by szhorvat** [comment]
Why does `fricas` have `supported_archs     i386 x86_64`? This line was added before Apple Silicon came along, so it can't be because of that? It causes inconvenience as trying to install fricas on an arm64 system now forcibly changes many installed ports to `+universal` (causing some to build from source), yet installation fails anyway with "Cannot install fricas for the arch 'x86_64' because its dependency sbcl does not build for the required arch by default". Can you look into this and fix fricas on arm64 if possible?

**Comment by pietvo** [comment]
> Why does `fricas` have `supported_archs i386 x86_64`? This line was added before Apple Silicon came along, so it can't be because of that? It causes inconvenience as trying to install fricas on an arm64 system now forcibly changes many installed ports to `+universal` (causing some to build from source), yet installation fails anyway with "Cannot install fricas for the arch 'x86_64' because its dependency sbcl does not build for the required arch by default". Can you look into this and fix fricas on arm64 if possible?

Probably my fault. @catap Can you take it out, please?

**Comment by szhorvat** [comment]
@pietvo I'd also like to make the suggestion to _not_ make the `+hunchentoot` variant default for `fricas`, with the following reasoning:

This variant causes the port to be non-distributable, so people need to build it from sources no matter what. Without this variant, it is distributable.

```
$ ./port_binary_distributable.tcl -v fricas
"fricas" is not distributable because its dependency "cl-puri" has license "llgpl" which is not known to be distributable

$ ./port_binary_distributable.tcl -v fricas -hunchentoot
"fricas" is distributable
```

If this variant is not enabled by default, then we could have quick-to-install binaries for `fricas`. That's a gain. People who need this variant would compile it themselves, just like before. Nothing lost.

**Comment by catap** [comment]
@szhorvat but this is requries for fricas integration into jupyter which I'm using ;)

**Comment by catap** [comment]
@pietvo ok, but not right now, a bit later

**Comment by catap** [comment]
@szhorvat BTW I'll double check license of cl-puri, maybe it has the second license which solves that issue.

**Comment by szhorvat** [comment]
> @szhorvat but this is requries for fricas integration into jupyter which I'm using ;)

You can still choose to install that variant on your system, no? Why make the installation experience painful by default, even for people who don't need this? Am I missing something?


> BTW I'll double check license of cl-puri, maybe it has the second license which solves that issue.

That would of course be great, but the chance is that there'll be many other license conflicts ...

**Comment by szhorvat** [comment]
> @szhorvat BTW I'll double check license of cl-puri, maybe it has the second license which solves that issue.

If you do this, perhaps you could have a look at `clisp` as well, which also gets built from sources when trying to install `fricas +hunchentoot`?

```
$ ./port_binary_distributable.tcl -v clisp
"clisp" is not distributable because its license "GPL-2" conflicts with license "AGPL-3" of dependency "ghostscript"
```

 - Should `ghostscript` really be a blocker? Is there linking to Ghostscript, or does it just call the executable (e.g. for doc building)?
 - Is `clisp` really just `GPL-2` and not `GPL-2+`? Unfortunately getting the answer to such questions can be a major pain.

**Comment by pietvo** [comment]
> @pietvo I'd also like to make the suggestion to _not_ make the `+hunchentoot` variant default for `fricas` ...

This change was made by Kirill (@catap), for reasons he explained here. But I actually agree that it would be better not to make it the default. Installing `hunchentoot` also installs `ecl` and `clisp` even if you are not using this. I don't know if there is a way to install `hunchentoot` with only the lisp that is used for `fricas`. That would help a bit, but on the other hand, if you want to have it you can easily use the `+hunchentoot` variant.

**Comment by catap** [comment]
agrees what removing `hunchentoot` from default variants seems good idea, will do it right now

**Comment by szhorvat** [comment]
FYI, `fricas` builds successfully on my arm64 machine if I remove `supported_archs     i386 x86_64` from the portfile (or perhaps better to just add `arm64`), and the test suite passes. I hope this information is helpful.

**Comment by catap** [comment]
ops, I've missed this remarks.

**Comment by catap** [comment]
> FYI, `fricas` builds successfully on my arm64 machine if I remove `supported_archs i386 x86_64` from the portfile (or perhaps better to just add `arm64`?), and the test suite passes. I hope this information is helpful.

better to remove it. support of PowerPC and i386 were rewerted to SBCL => fricas can be build everywhere with probability almost 1 I assume

**Comment by pmetzger** [comment]
@catap Is this now ready to merge?

**Comment by catap** [comment]
@pmetzger yes, it is

**Comment by barracuda156** [comment]
@catap SBCL 2.3.9 confirmed to build fine on PowerPC!

**Pull request merged**